### PR TITLE
[function] Add support for inet_aton

### PR DESCRIPTION
**Pull request by mshauneu**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Add support for inet_aton(convert IPv4 string to bytes)

## Changelog

- New Feature

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
`make lint` command will help here :D

File: common/functions/src/scalars/others/inet_aton.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            .column()
+            .to_minimal_array()?
+            .cast_with_type(&DataType::String)?;
+
+        let opt_iter = opt_iter.string()?.into_iter().map(|val_opt| {
+            val_opt.map(|v| {
+                u32::from(
+                    String::from_utf8_lossy(v)
+                        .parse::<std::net::Ipv4Addr>()
+                        .unwrap_or(std::net::Ipv4Addr::new(0, 0, 0, 0)),
```
Comment (sundy-li) [inline common/functions/src/scalars/others/inet_aton.rs:85]: Maybe unwrap should be none.

```
MySQL > select inet_aton('127.0.0.x22');
+--------------------------+
| inet_aton('127.0.0.x22') | 
+--------------------------+
| <nil>                    | 
+--------------------------+
1 row in set (0.00 sec)

```

Comment (mshauneu) [reply common/functions/src/scalars/others/inet_aton.rs]: Fixed:
```
mysql> SELECT inet_aton('127.0.0.');
+-----------------------+
| inet_aton('127.0.0.') |
+-----------------------+
|                  NULL |
+-----------------------+
```

**Review by sundy-li — APPROVED** [review]

**Pull request merged**